src/inputType/RotatePanInput.ts

```diff
@@ -0,0 +1,136 @@
+import Axes from "../Axes";
+import { toAxis } from "./InputType";
+import { PanInput, PanInputOption } from "./PanInput";
+
+export class RotatePanInput extends PanInput {
+	private rotateOrigin: number[];
+	private prevAngle: number;
+	private prevQuadrant: number;
+	private lastDiff: number;
+	private coefficientForDistanceToAngle: number;
+
+	constructor(el: string | HTMLElement, options?: PanInputOption) {
+		super(el, options);
+
+		const rect = this.element.getBoundingClientRect();
+		// TODO: how to do if element is ellipse not circle.
+		this.coefficientForDistanceToAngle = 360 / (rect.width * Math.PI); // from 2*pi*r * x / 360
+		// TODO: provide a way to set origin like https://developer.mozilla.org/en-US/docs/Web/CSS/transform-origin
+		this.rotateOrigin = [rect.left + (rect.width - 1) / 2, rect.top + (rect.height - 1) / 2];
+		this.prevAngle = null;
+		this.prevQuadrant = null;
+		this.lastDiff = 0;
+	}
+
+	mapAxes(axes: string[]) {
+		this._direction = Axes.DIRECTION_ALL;
+		this.axes = axes;
+	}
+
+	onHammerInput(event) {
+		if (this.isEnable()) {
+			if (event.isFirst) {
+				this.prevAngle = null;
+				this.observer.hold(this, event);
+				this.onPanstart(event);
+			} else if (event.isFinal) {
+				this.onPanend(event);
+			}
+		}
+	}
+
+	onPanstart(event) {
+		this.triggerChange(event);
+	}
+
+	onPanmove(event) {
+		this.triggerChange(event);
+	}
+
+	onPanend(event) {
+		this.triggerChange(event);
+		this.triggerAnimation(event);
+	}
+
+	private triggerChange(event) {
+		const angle = this.getAngle(event.center.x, event.center.y);
+		const quadrant = this.getQuadrant(event.center.x, event.center.y);
+		const diff = this.getDifference(this.prevAngle, angle, this.prevQuadrant, quadrant);
```

Comment:
@jongmoon

Why don't we combine getAngle, getQuadrant, and getDifferences into one method?

Reply:
I make method only if that is called twice. But they are called once.